saleor/checkout/calculations.py

```diff
@@ -355,22 +328,52 @@ def _calculate_and_add_tax(
     address: Optional["Address"] = None,
 ):
     if tax_calculation_strategy == TaxCalculationStrategy.TAX_APP:
-        # Call the tax plugins.
+        # If taxAppId is not configured remain previous behaviour to call plugin and apps
+        if not tax_app_identifier:
+            # Call the tax plugins.
+            _apply_tax_data_from_plugins(
+                checkout, manager, checkout_info, lines, address
+            )
+            # Get the taxes calculated with apps and apply to checkout.
+            tax_data = manager.get_taxes_for_checkout(
+                checkout_info, lines, tax_app_identifier
+            )
+            _apply_tax_data(checkout, lines, tax_data)
+        else:
+            _call_plugin_or_app_tax(
+                tax_app_identifier,
+                checkout,
+                manager,
+                checkout_info,
+                lines,
+                address,
+            )
+    else:
+        # Get taxes calculated with flat rates and apply to checkout.
+        update_checkout_prices_with_flat_rates(
+            checkout, checkout_info, lines, prices_entered_with_tax, address
+        )
+
+
+def _call_plugin_or_app_tax(
+    tax_app_identifier: Optional[str],
+    checkout: "Checkout",
+    manager: "PluginsManager",
+    checkout_info: "CheckoutInfo",
+    lines: Iterable["CheckoutLineInfo"],
+    address: Optional["Address"] = None,
+):
+    from ..plugins.avatax.plugin import AvataxPlugin
+
+    if tax_app_identifier == AvataxPlugin.PLUGIN_IDENTIFIER:
```

Comment:
I don't think that this is correct approach. You're focusing here directly on AvataxPlugin. If you see in the code, you will see that all (or almost all) communication between Saleor and Plugins goes through the PluginsManager. 
With this approach, you will break any potential custom tax plugins in case when someone will set the plugin identifier as tax app. 
Instead of this, you should have a taxPlugin ID returned in form like (just an example): "taxPlugin:<plugin-identifier>". This should be passed to the plugin manager, and plugin manager should extract the plugin identifier and call the logic on the expected plugin (if exists). 

---

Additionally I think, that you don't need to modify the logic responsible for handling the taxes here. 
You could have something like this:
```python
def _calculate_and_add_tax(
    tax_calculation_strategy: str,
    tax_app_identifier: Optional[str],
    checkout: "Checkout",
    manager: "PluginsManager",
    checkout_info: "CheckoutInfo",
    lines: Iterable["CheckoutLineInfo"],
    prices_entered_with_tax: bool,
    address: Optional["Address"] = None,
    tax_identifier: Optional[str] = None,
):
    if tax_calculation_strategy == TaxCalculationStrategy.TAX_APP:
        # Call the tax plugins.
        _apply_tax_data_from_plugins(checkout, manager, checkout_info, lines, address, tax_identifier)
        # Get the taxes calculated with apps and apply to checkout.
        tax_data = manager.get_taxes_for_checkout(
            checkout_info, lines, tax_app_identifier
        )
        # If taxAppId is not configured we will for now allow to finalize process for
        # backward compatibility.
        if tax_data is None and tax_app_identifier is not None:
            raise EmptyTaxData("Empty tax data.")
        _apply_tax_data(checkout, lines, tax_data)
    else:
        # Get taxes calculated with flat rates and apply to checkout.
        update_checkout_prices_with_flat_rates(
            checkout, checkout_info, lines, prices_entered_with_tax, address
        )
```
(just added `tax_identifier` as argument), then you can pass it to each manager call, and make something like:
```python
    def calculate_checkout_line_total(
        self,
        checkout_info: "CheckoutInfo",
        lines: Iterable["CheckoutLineInfo"],
        checkout_line_info: "CheckoutLineInfo",
        address: Optional["Address"],
            tax_identifier: Optional[str]
    ) -> TaxedMoney:
    ...
        if tax_identifier: 
            if not tax_identifier.startswith("taxPlugin"):
                return default_taxed_value
            else:
                tax_identifier = tax_identifier.split("taxPlugin")[1]
                return self.__run_method_on_single_plugin()

        line_total = self.__run_method_on_plugins(
```

---

Another way of handling it in calculations:
```python
def _calculate_and_add_tax(
    tax_calculation_strategy: str,
    tax_app_identifier: Optional[str],
    checkout: "Checkout",
    manager: "PluginsManager",
    checkout_info: "CheckoutInfo",
    lines: Iterable["CheckoutLineInfo"],
    prices_entered_with_tax: bool,
    address: Optional["Address"] = None,
    tax_identifier: Optional[str] = None,
):
    if tax_calculation_strategy == TaxCalculationStrategy.TAX_APP:
        if tax_app_identifier and tax_identifier.startswith("taxPlugin:"):
            # Call the tax plugins.
            _apply_tax_data_from_plugins(checkout, manager, checkout_info, lines, address, tax_identifier)
        # Get the taxes calculated with apps and apply to checkout.
        tax_data = manager.get_taxes_for_checkout(
            checkout_info, lines, tax_app_identifier
        )
```

Code snippets are only suggestions. The most important thing is that we don't want to have any custom logic for plugins inside the direct Saleor code. Saleor doesn't know anything about plugins. Saloer knows only PluginsManager

Reply:
Remove direct usage of `AvataxPlugin`

Regarding the code snippets I decided to leave it, as it is, cause I believe it's easier to read code and logic regarding circuit break is more compact.